vignettes/datatable-joins.Rmd

```diff
@@ -0,0 +1,687 @@
+  - The *second group* of columns in the new data.table comes from the `i` table.
+  - If the join operation presents a present any **name conflict** (both table have same column name) the ***prefix*** `i.` is added to column names from the **right-hand table** (table on `i` position).
+  
+- **Row level**
+  - The missing `product_id` present on the `ProductReceived` table in row 1 was successfully matched with missing `id` of the `Products` table, so `NA` ***values are treated as any other value***.
+  - All rows from in the `i` table were kept including:
+    - Not matching rows like the one with `product_id = 6`.
+    - Rows that repeat the same `product_id` several times.
+    
+#### 3.1.1. Joining by a list argument
+
+If you are following the vignette, you might have found out that we used a vector to define the relations between tables in the `on` argument, that is really useful if you are **creating your own functions**, but another alternative is to use a **list** to define the columns to match.
+
+To use this capacity, we have 2 equivalent alternatives:
+
+- Wrapping the related columns in the base R `list` function.
+
+```{r, eval=FALSE}
+Products[ProductReceived,
+         on = list(id = product_id)]
+```
+
+- Wrapping the related columns in the data.table `list`	alias `.`.
+
+```{r, eval=FALSE}
+Products[ProductReceived,
+         on = .(id = product_id)]
+```
+
+#### 3.1.2. Alternatives to define the `on` argument
+
+In all the prior example we have pass the column names we want to match to the `on` argument but `data.table` also have alternatives to that syntax.
+
+- **Natural join**: Selects the columns to perform the match based on common column names. To illustrate this method, let's change the column of `Products` table from `id` to `product_id` and use the keyword `.NATURAL`.
+
+```{r}
+ProductsChangedName = setnames(copy(Products), "id", "product_id")
+ProductsChangedName
+
+ProductsChangedName[ProductReceived, on = .NATURAL]
+```
+
+- **Keyed join**: Selects the columns to perform the match based on keyed columns regardless of their names.To illustrate this method, we need to define keys in the same order for both tables.
+
+```{r}
+ProductsKeyed = setkey(copy(Products), id)
+key(ProductsKeyed)
+
+ProductReceivedKeyed = setkey(copy(ProductReceived), product_id)
+key(ProductReceivedKeyed)
+
+ProductsKeyed[ProductReceivedKeyed]
+```
+
+#### 3.1.3. Operations after joining
+
+Most of the time after a join is complete we need to make some additional transformations. To make so we have the following alternatives:
+
+- Chaining a new instruction by adding a pair of brakes `[]`.
+- Passing a list with the columns that we want to keep or create to the `j` argument.
+
+Our recommendation is to use the second alternative if possible, as it is **faster** and uses **less memory** than the first one.
+
+
+##### 3.1.3.1. Managing shared column Names with the j argument
```

Comment:
four levels of section numbering is difficult to understand. can it be kept to 2 or 3 please?